2023-03-30 15:20:05,031 - INFO - [MainThread] - Initialized router P2 at port 5002 PID 102925
2023-03-30 15:20:05,033 - INFO - [MainThread] - Adding a connection to 4mnesia
2023-03-30 15:20:05,033 - INFO - [Connect-4mnesia] - Calling connect_to_rendezvous_point
2023-03-30 15:20:05,033 - INFO - [Connect-4mnesia] - Connect to rendezvous point 4mnesia
2023-03-30 15:20:05,036 - INFO - [MainThread] - Adding listener for LupusreginaBeta
2023-03-30 15:20:05,036 - INFO - [Establish-LupusreginaBeta] - Establishing RP to LupusreginaBeta
2023-03-30 15:20:05,039 - INFO - [MainThread] - Starting testing thread
2023-03-30 15:20:05,040 - INFO - [TESTER] - Started tester
2023-03-30 15:20:05,054 - INFO - [MainThread] - Starting connection thread
2023-03-30 15:20:05,056 - INFO - [Connect-4mnesia] - Loading cached NetworkStatusDocument from TorCacheDirStorage: /root/.local/share/torpy/network_status
2023-03-30 15:20:05,056 - INFO - [MainThread] - Starting log thread
2023-03-30 15:20:05,082 - INFO - [Establish-LupusreginaBeta] - Loading cached NetworkStatusDocument from TorCacheDirStorage: /root/.local/share/torpy/network_status
2023-03-30 15:20:05,636 - INFO - [Connect-4mnesia] - Loading cached DirKeyCertificateList from TorCacheDirStorage: /root/.local/share/torpy/dir_key_certificates
2023-03-30 15:20:05,637 - INFO - [Connect-4mnesia] - Downloading new consensus...
2023-03-30 15:20:05,637 - INFO - [Connect-4mnesia] - Connecting to guard node 5.9.158.75:9001... (Internal dir client)
2023-03-30 15:20:05,673 - INFO - [Establish-LupusreginaBeta] - Loading cached DirKeyCertificateList from TorCacheDirStorage: /root/.local/share/torpy/dir_key_certificates
2023-03-30 15:20:05,674 - INFO - [Establish-LupusreginaBeta] - Downloading new consensus...
2023-03-30 15:20:05,674 - INFO - [Establish-LupusreginaBeta] - Connecting to guard node 192.87.28.28:9001... (Internal dir client)
2023-03-30 15:20:05,705 - ERROR - [Connect-4mnesia] - [Errno 111] Connection refused
2023-03-30 15:20:05,705 - ERROR - [Connect-4mnesia] - [ignored]
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cell_socket.py", line 63, in connect
    self._socket.connect((self._router.ip, self._router.or_port))
  File "/usr/lib/python3.9/ssl.py", line 1342, in connect
    self._real_connect(addr, False)
  File "/usr/lib/python3.9/ssl.py", line 1329, in _real_connect
    super().connect(addr)
ConnectionRefusedError: [Errno 111] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/utils.py", line 79, in newfn
    return func(*args, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/consesus.py", line 235, in renew
    raw_string = self.download_consensus(prev_hash)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/consesus.py", line 183, in newfn
    return func(*args, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/consesus.py", line 395, in download_consensus
    with self._get_dir_client() as dir_client:
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/consesus.py", line 380, in _get_dir_client
    self._dir_guard, self._dir_circuit = self._create_dir_circuit(purpose='Internal dir client')
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/consesus.py", line 370, in _create_dir_circuit
    guard = TorGuard(router, purpose=purpose)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 66, in __init__
    self.__tor_socket.connect()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cell_socket.py", line 69, in connect
    raise TorSocketConnectError(e)
torpy.cell_socket.TorSocketConnectError: [Errno 111] Connection refused
2023-03-30 15:20:05,706 - WARNING - [Connect-4mnesia] - Retry with another router...
2023-03-30 15:20:05,706 - INFO - [Connect-4mnesia] - Wait 3 sec before next retry
2023-03-30 15:20:05,811 - INFO - [Establish-LupusreginaBeta] - Creating new circuit #80000001 with 192.87.28.28:9001 router...
2023-03-30 15:20:05,848 - INFO - [Establish-LupusreginaBeta] - Building 0 hops circuit...
2023-03-30 15:20:05,848 - INFO - [Establish-LupusreginaBeta] - Stream #1: creating attached to #80000001 circuit...
2023-03-30 15:20:05,848 - INFO - [Establish-LupusreginaBeta] - Stream #1: connecting to hsdir
2023-03-30 15:20:05,885 - INFO - [RecvLoop_192.87.] - Stream #1: connected (remote ip '')
2023-03-30 15:20:06,560 - INFO - [RecvLoop_192.87.] - Stream #1: remote disconnected (reason = DONE)
2023-03-30 15:20:06,567 - INFO - [Establish-LupusreginaBeta] - Stream #1: closing (state = Disconnected)...
2023-03-30 15:20:06,798 - WARNING - [Establish-LupusreginaBeta] - Unknown voter present
2023-03-30 15:20:06,811 - INFO - [Establish-LupusreginaBeta] - Connecting to guard node 185.181.61.23:443 (staysafeandgoodluck; Tor 0.4.7.10)... (None)
2023-03-30 15:20:06,861 - INFO - [Establish-LupusreginaBeta] - Creating new circuit #80000002 with 185.181.61.23:443 (staysafeandgoodluck; Tor 0.4.7.10) router...
2023-03-30 15:20:06,875 - INFO - [Establish-LupusreginaBeta] - Building -1 hops circuit...
2023-03-30 15:20:06,875 - INFO - [Establish-LupusreginaBeta] - Extending the circuit #80000002 with 213.113.2.72:6881 (LupusreginaBeta; Tor 0.4.7.13)...
2023-03-30 15:20:06,876 - INFO - [Establish-LupusreginaBeta] - Getting descriptor for 213.113.2.72:6881 (LupusreginaBeta; Tor 0.4.7.13)...
2023-03-30 15:20:06,876 - INFO - [Establish-LupusreginaBeta] - Stream #2: creating attached to #80000001 circuit...
2023-03-30 15:20:06,877 - INFO - [Establish-LupusreginaBeta] - Stream #2: connecting to hsdir
2023-03-30 15:20:06,918 - INFO - [RecvLoop_192.87.] - Stream #2: connected (remote ip '')
2023-03-30 15:20:07,117 - INFO - [RecvLoop_192.87.] - Stream #2: remote disconnected (reason = DONE)
2023-03-30 15:20:07,118 - INFO - [Establish-LupusreginaBeta] - Stream #2: closing (state = Disconnected)...
2023-03-30 15:20:07,118 - INFO - [Establish-LupusreginaBeta] - Got descriptor
2023-03-30 15:20:07,147 - INFO - [Establish-LupusreginaBeta] - <class 'torpy.circuit.TorCircuit'>
2023-03-30 15:20:07,174 - INFO - [Establish-LupusreginaBeta] - Rendezvous established (CellRelayRendezvousEstablished())
2023-03-30 15:20:07,174 - INFO - [Establish-LupusreginaBeta] - Waiting for connections at relay LupusreginaBeta for cookie b'eb688e4f52df90278060' ...
2023-03-30 15:20:08,709 - INFO - [Connect-4mnesia] - Downloading new consensus...
2023-03-30 15:20:08,710 - INFO - [Connect-4mnesia] - Connecting to guard node 108.53.208.157:443... (Internal dir client)
2023-03-30 15:20:09,134 - INFO - [Connect-4mnesia] - Creating new circuit #80000003 with 108.53.208.157:443 router...
2023-03-30 15:20:09,235 - INFO - [Connect-4mnesia] - Building 0 hops circuit...
2023-03-30 15:20:09,235 - INFO - [Connect-4mnesia] - Stream #3: creating attached to #80000003 circuit...
2023-03-30 15:20:09,235 - INFO - [Connect-4mnesia] - Stream #3: connecting to hsdir
2023-03-30 15:20:09,338 - INFO - [RecvLoop_108.53.] - Stream #3: connected (remote ip '')
2023-03-30 15:20:10,921 - INFO - [RecvLoop_108.53.] - Stream #3: remote disconnected (reason = DONE)
2023-03-30 15:20:10,929 - INFO - [Connect-4mnesia] - Stream #3: closing (state = Disconnected)...
2023-03-30 15:20:11,165 - WARNING - [Connect-4mnesia] - Unknown voter present
2023-03-30 15:20:11,541 - INFO - [Connect-4mnesia] - Connected to rendezvous point 4mnesia
2023-03-30 15:20:11,541 - INFO - [Connect-4mnesia] - Waiting for peer to open stream
2023-03-30 15:20:11,644 - INFO - [Connect-4mnesia] - Received cell: <class 'torpy.cells.CellDestroy'>
2023-03-30 15:20:11,644 - WARNING - [Connect-4mnesia] - Could not connect to rendezvous point 4mnesia: Receive CellDestroy from RP, 11 attempts left...
2023-03-30 15:20:14,064 - INFO - [Establish-LupusreginaBeta] - Got REND2 message from LupusreginaBeta
2023-03-30 15:20:14,064 - INFO - [Establish-LupusreginaBeta] - Entered circuit context with id 0x80000002
2023-03-30 15:20:14,064 - INFO - [Establish-LupusreginaBeta] - Stream #4: creating attached to #80000002 circuit...
2023-03-30 15:20:14,064 - INFO - [Establish-LupusreginaBeta] - Stream #4: connecting to ('127.0.0.1', 5000)
2023-03-30 15:20:14,121 - INFO - [RecvLoop_185.181] - Stream #4: connected (remote ip '127.0.0.1')
2023-03-30 15:20:14,266 - INFO - [Establish-LupusreginaBeta] - Sending HELLO to peer
2023-03-30 15:20:14,266 - INFO - [Establish-LupusreginaBeta] - Sending HELLO to peer: b'\x80\x04\x95&\x00\x00\x00\x00\x00\x00\x00]\x94}\x94(\x8c\x07Message\x94\x8c\x05HELLO\x94\x8c\x04Peer\x94\x8c\x02P2\x94ua.'
2023-03-30 15:20:14,267 - INFO - [RECEIVER] - Added neighbour P4
2023-03-30 15:20:16,646 - INFO - [Connect-4mnesia] - Calling connect_to_rendezvous_point
2023-03-30 15:20:16,647 - INFO - [Connect-4mnesia] - Connect to rendezvous point 4mnesia
2023-03-30 15:20:16,650 - INFO - [Connect-4mnesia] - Loading cached NetworkStatusDocument from TorCacheDirStorage: /root/.local/share/torpy/network_status
2023-03-30 15:20:16,886 - INFO - [Connect-4mnesia] - Loading cached DirKeyCertificateList from TorCacheDirStorage: /root/.local/share/torpy/dir_key_certificates
2023-03-30 15:20:17,155 - INFO - [Connect-4mnesia] - Connected to rendezvous point 4mnesia
2023-03-30 15:20:17,155 - INFO - [Connect-4mnesia] - Waiting for peer to open stream
2023-03-30 15:20:17,210 - INFO - [Connect-4mnesia] - Received cell: <class 'torpy.cells.CellDestroy'>
2023-03-30 15:20:17,210 - WARNING - [Connect-4mnesia] - Could not connect to rendezvous point 4mnesia: Receive CellDestroy from RP, 10 attempts left...
2023-03-30 15:20:22,216 - INFO - [Connect-4mnesia] - Calling connect_to_rendezvous_point
2023-03-30 15:20:22,216 - INFO - [Connect-4mnesia] - Connect to rendezvous point 4mnesia
2023-03-30 15:20:22,219 - INFO - [Connect-4mnesia] - Loading cached NetworkStatusDocument from TorCacheDirStorage: /root/.local/share/torpy/network_status
2023-03-30 15:20:22,451 - INFO - [Connect-4mnesia] - Loading cached DirKeyCertificateList from TorCacheDirStorage: /root/.local/share/torpy/dir_key_certificates
2023-03-30 15:20:22,722 - INFO - [Connect-4mnesia] - Connected to rendezvous point 4mnesia
2023-03-30 15:20:22,723 - INFO - [Connect-4mnesia] - Waiting for peer to open stream
2023-03-30 15:20:22,784 - INFO - [Connect-4mnesia] - Received cell: <class 'torpy.cells.CellDestroy'>
2023-03-30 15:20:22,784 - WARNING - [Connect-4mnesia] - Could not connect to rendezvous point 4mnesia: Receive CellDestroy from RP, 9 attempts left...
2023-03-30 15:20:27,788 - INFO - [Connect-4mnesia] - Calling connect_to_rendezvous_point
2023-03-30 15:20:27,789 - INFO - [Connect-4mnesia] - Connect to rendezvous point 4mnesia
2023-03-30 15:20:27,793 - INFO - [Connect-4mnesia] - Loading cached NetworkStatusDocument from TorCacheDirStorage: /root/.local/share/torpy/network_status
2023-03-30 15:20:28,030 - INFO - [Connect-4mnesia] - Loading cached DirKeyCertificateList from TorCacheDirStorage: /root/.local/share/torpy/dir_key_certificates
2023-03-30 15:20:28,302 - INFO - [Connect-4mnesia] - Connected to rendezvous point 4mnesia
2023-03-30 15:20:28,302 - INFO - [Connect-4mnesia] - Waiting for peer to open stream
2023-03-30 15:20:28,453 - INFO - [Connect-4mnesia] - Received cell: <class 'torpy.cells.CellRelay'>
2023-03-30 15:20:28,454 - INFO - [Connect-4mnesia] - Stream opened successfully
2023-03-30 15:20:28,454 - INFO - [Connect-4mnesia] - Sending HELLO to peer: b'\x80\x04\x95&\x00\x00\x00\x00\x00\x00\x00]\x94}\x94(\x8c\x07Message\x94\x8c\x05HELLO\x94\x8c\x04Peer\x94\x8c\x02P2\x94ua.'
2023-03-30 15:20:28,605 - INFO - [RECEIVER] - Added neighbour P1
2023-03-30 15:21:08,838 - INFO - [RecvLoop_192.87.] - On destroy: circuit #80000001
2023-03-30 15:21:08,838 - INFO - [RecvLoop_192.87.] - Destroy circuit #80000001
2023-03-30 15:21:15,406 - INFO - [RecvLoop_108.53.] - On destroy: circuit #80000003
2023-03-30 15:21:15,406 - INFO - [RecvLoop_108.53.] - Destroy circuit #80000003
2023-03-30 15:21:26,376 - WARNING - [Connect-4mnesia] - Got SENDME
2023-03-30 15:22:25,356 - WARNING - [Connect-4mnesia] - Got SENDME
2023-03-30 15:22:25,398 - WARNING - [Connect-4mnesia] - Got SENDME
2023-03-30 15:23:26,438 - WARNING - [Connect-4mnesia] - Got SENDME
2023-03-30 15:24:50,558 - WARNING - [Connect-4mnesia] - Got SENDME
2023-03-30 15:24:50,597 - WARNING - [Connect-4mnesia] - Got SENDME
2023-03-30 15:26:05,371 - INFO - [CONNECTION] - Loading cached NetworkStatusDocument from TorCacheDirStorage: /root/.local/share/torpy/network_status
2023-03-30 15:26:06,266 - INFO - [CONNECTION] - Loading cached DirKeyCertificateList from TorCacheDirStorage: /root/.local/share/torpy/dir_key_certificates
2023-03-30 15:26:06,284 - INFO - [MainThread] - ['ESTABLISH', 'IdeaNexus', b'ccec1e753f5d61493760']
2023-03-30 15:26:06,284 - INFO - [MainThread] - Got ESTABLISH to relay IdeaNexus:b'ccec1e753f5d61493760' via staysafeandgoodluck
2023-03-30 15:26:06,286 - INFO - [Establish-IdeaNexus] - Establishing RP to IdeaNexus
2023-03-30 15:26:06,343 - INFO - [Establish-IdeaNexus] - Loading cached NetworkStatusDocument from TorCacheDirStorage: /root/.local/share/torpy/network_status
2023-03-30 15:26:07,273 - INFO - [Establish-IdeaNexus] - Loading cached DirKeyCertificateList from TorCacheDirStorage: /root/.local/share/torpy/dir_key_certificates
2023-03-30 15:26:07,276 - INFO - [Establish-IdeaNexus] - Connecting to guard node 185.181.61.23:443 (staysafeandgoodluck; Tor 0.4.7.10)... (None)
2023-03-30 15:26:07,341 - INFO - [Establish-IdeaNexus] - Creating new circuit #80000004 with 185.181.61.23:443 (staysafeandgoodluck; Tor 0.4.7.10) router...
2023-03-30 15:26:07,362 - INFO - [Establish-IdeaNexus] - Building -1 hops circuit...
2023-03-30 15:26:07,363 - INFO - [Establish-IdeaNexus] - Extending the circuit #80000004 with 164.68.121.40:9001 (IdeaNexus; Tor 0.4.7.13)...
2023-03-30 15:26:07,364 - INFO - [Establish-IdeaNexus] - Getting descriptor for 164.68.121.40:9001 (IdeaNexus; Tor 0.4.7.13)...
2023-03-30 15:26:07,372 - INFO - [Establish-IdeaNexus] - Connecting to guard node 178.175.148.161:9001 (alex; Tor 0.4.7.13)... (Internal dir client)
2023-03-30 15:26:11,793 - WARNING - [Connect-4mnesia] - Got SENDME
2023-03-30 15:26:12,391 - INFO - [RECEIVER] - Added relay WinstonSmith from P1 to JOIN queue
2023-03-30 15:26:12,391 - INFO - [MainThread] - ['JOIN', 'WinstonSmith', b'ccec1e753f5d61493760']
2023-03-30 15:26:12,392 - INFO - [MainThread] - Got JOIN to relay WinstonSmith, waiting 10 sec to connect
2023-03-30 15:26:12,397 - INFO - [Connect-WinstonSmith] - Calling connect_to_rendezvous_point
2023-03-30 15:26:12,397 - INFO - [Connect-WinstonSmith] - Connect to rendezvous point WinstonSmith
2023-03-30 15:26:12,454 - INFO - [Connect-WinstonSmith] - Loading cached NetworkStatusDocument from TorCacheDirStorage: /root/.local/share/torpy/network_status
2023-03-30 15:26:13,331 - INFO - [Connect-WinstonSmith] - Loading cached DirKeyCertificateList from TorCacheDirStorage: /root/.local/share/torpy/dir_key_certificates
2023-03-30 15:26:13,600 - INFO - [Connect-WinstonSmith] - Connected to rendezvous point WinstonSmith
2023-03-30 15:26:13,602 - INFO - [Connect-WinstonSmith] - Waiting for peer to open stream
2023-03-30 15:26:13,705 - INFO - [Connect-WinstonSmith] - Received cell: <class 'torpy.cells.CellRelay'>
2023-03-30 15:26:13,706 - INFO - [Connect-WinstonSmith] - Stream opened successfully
2023-03-30 15:26:13,707 - INFO - [Connect-WinstonSmith] - Sending HELLO to peer: b'\x80\x04\x95&\x00\x00\x00\x00\x00\x00\x00]\x94}\x94(\x8c\x07Message\x94\x8c\x05HELLO\x94\x8c\x04Peer\x94\x8c\x02P2\x94ua.'
2023-03-30 15:26:13,916 - INFO - [RECEIVER] - Added neighbour P1
2023-03-30 15:26:16,295 - INFO - [CONNECTION] - Sending JOIN to P4 for relay IdeaNexus with cookie b'ccec1e753f5d61493760'
2023-03-30 15:26:22,389 - ERROR - [Establish-IdeaNexus] - timed out
2023-03-30 15:26:22,390 - ERROR - [Establish-IdeaNexus] - [ignored]
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cell_socket.py", line 63, in connect
    self._socket.connect((self._router.ip, self._router.or_port))
  File "/usr/lib/python3.9/ssl.py", line 1342, in connect
    self._real_connect(addr, False)
  File "/usr/lib/python3.9/ssl.py", line 1329, in _real_connect
    super().connect(addr)
socket.timeout: timed out

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/utils.py", line 79, in newfn
    return func(*args, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/consesus.py", line 183, in newfn
    return func(*args, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/consesus.py", line 431, in get_descriptor
    with self._get_dir_client() as dir_client:
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/consesus.py", line 380, in _get_dir_client
    self._dir_guard, self._dir_circuit = self._create_dir_circuit(purpose='Internal dir client')
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/consesus.py", line 370, in _create_dir_circuit
    guard = TorGuard(router, purpose=purpose)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 66, in __init__
    self.__tor_socket.connect()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cell_socket.py", line 69, in connect
    raise TorSocketConnectError(e)
torpy.cell_socket.TorSocketConnectError: timed out
2023-03-30 15:26:22,390 - WARNING - [Establish-IdeaNexus] - Retry with another router...
2023-03-30 15:26:22,390 - INFO - [Establish-IdeaNexus] - Wait 1 sec before next retry
2023-03-30 15:26:23,397 - INFO - [Establish-IdeaNexus] - Connecting to guard node 93.115.26.40:9051 (Unnamed; Tor 0.4.7.12)... (Internal dir client)
2023-03-30 15:26:23,560 - INFO - [Establish-IdeaNexus] - Creating new circuit #80000005 with 93.115.26.40:9051 (Unnamed; Tor 0.4.7.12) router...
2023-03-30 15:26:23,599 - INFO - [Establish-IdeaNexus] - Building 0 hops circuit...
2023-03-30 15:26:23,599 - INFO - [Establish-IdeaNexus] - Stream #5: creating attached to #80000005 circuit...
2023-03-30 15:26:23,599 - INFO - [Establish-IdeaNexus] - Stream #5: connecting to hsdir
2023-03-30 15:26:23,644 - INFO - [RecvLoop_93.115.] - Stream #5: connected (remote ip '')
2023-03-30 15:26:23,847 - INFO - [RecvLoop_93.115.] - Stream #5: remote disconnected (reason = DONE)
2023-03-30 15:26:23,848 - INFO - [Establish-IdeaNexus] - Stream #5: closing (state = Disconnected)...
2023-03-30 15:26:23,849 - INFO - [Establish-IdeaNexus] - Got descriptor
2023-03-30 15:26:23,939 - INFO - [Establish-IdeaNexus] - <class 'torpy.circuit.TorCircuit'>
2023-03-30 15:26:23,988 - INFO - [Establish-IdeaNexus] - Rendezvous established (CellRelayRendezvousEstablished())
2023-03-30 15:26:23,988 - INFO - [Establish-IdeaNexus] - Waiting for connections at relay IdeaNexus for cookie b'ccec1e753f5d61493760' ...
2023-03-30 15:26:29,601 - INFO - [Establish-IdeaNexus] - Got REND2 message from IdeaNexus
2023-03-30 15:26:29,606 - INFO - [Establish-IdeaNexus] - Entered circuit context with id 0x80000004
2023-03-30 15:26:29,606 - INFO - [Establish-IdeaNexus] - Stream #6: creating attached to #80000004 circuit...
2023-03-30 15:26:29,607 - INFO - [Establish-IdeaNexus] - Stream #6: connecting to ('127.0.0.1', 5000)
2023-03-30 15:26:29,709 - INFO - [RecvLoop_185.181] - Stream #6: connected (remote ip '127.0.0.1')
2023-03-30 15:26:29,810 - INFO - [Establish-IdeaNexus] - Sending HELLO to peer
2023-03-30 15:26:29,810 - INFO - [Establish-IdeaNexus] - Sending HELLO to peer: b'\x80\x04\x95&\x00\x00\x00\x00\x00\x00\x00]\x94}\x94(\x8c\x07Message\x94\x8c\x05HELLO\x94\x8c\x04Peer\x94\x8c\x02P2\x94ua.'
2023-03-30 15:26:29,811 - INFO - [RECEIVER] - Added neighbour P4
2023-03-30 15:27:10,745 - WARNING - [Connect-WinstonSmith] - Got SENDME
2023-03-30 15:27:11,805 - ERROR - [Connect-4mnesia] - Got DESTROY
2023-03-30 15:27:28,637 - ERROR - [Establish-LupusreginaBeta] - recv timeout
2023-03-30 15:27:28,637 - INFO - [Establish-LupusreginaBeta] - Closing connection Establish-LupusreginaBeta
2023-03-30 15:27:28,638 - INFO - [Establish-LupusreginaBeta] - Stream #4: closing (state = Connected)...
2023-03-30 15:27:28,639 - INFO - [Establish-LupusreginaBeta] - Destroy circuit #80000002
2023-03-30 15:27:28,639 - INFO - [Establish-LupusreginaBeta] - Closing guard connections (None)...
2023-03-30 15:27:34,544 - INFO - [RecvLoop_93.115.] - On destroy: circuit #80000005
2023-03-30 15:27:34,545 - INFO - [RecvLoop_93.115.] - Destroy circuit #80000005
2023-03-30 15:27:40,237 - INFO - [RECEIVER] - updateGraphOnly: [['P3', 'P4', 100, 'KeffalsIsNotAWoman'], ['P1', 'P3', 100, 'artikel10buc06'], ['P2', 'P4', 100, 'IdeaNexus'], ['P1', 'P2', 100, 'WinstonSmith']] , {'P1'}
2023-03-30 15:27:51,472 - INFO - [RECEIVER] - updateGraphOnly: [['P3', 'P4', 100, 'KeffalsIsNotAWoman'], ['P2', 'P4', 100, 'IdeaNexus']] , {'P3'}
2023-03-30 15:28:14,951 - WARNING - [Connect-WinstonSmith] - Got SENDME
2023-03-30 15:28:15,709 - WARNING - [Connect-WinstonSmith] - Got SENDME
2023-03-30 15:28:58,188 - WARNING - [RECEIVER] - Routing loop from P2 to P1: P2-P4-P2
2023-03-30 15:29:02,010 - WARNING - [RECEIVER] - Routing loop from P2 to P1: P2-P4-P2
2023-03-30 15:29:22,030 - WARNING - [Connect-WinstonSmith] - Got SENDME
2023-03-30 15:30:48,193 - WARNING - [Connect-WinstonSmith] - Got SENDME
2023-03-30 15:30:49,993 - WARNING - [Connect-WinstonSmith] - Got SENDME
2023-03-30 15:31:35,054 - INFO - [RECEIVER] - Added relay NTH17R3 from P1 to JOIN queue
2023-03-30 15:31:35,055 - INFO - [MainThread] - ['JOIN', 'NTH17R3', b'b6f0416f0dbf6dac1c05']
2023-03-30 15:31:35,056 - INFO - [MainThread] - Got JOIN to relay NTH17R3, waiting 10 sec to connect
2023-03-30 15:31:35,056 - INFO - [Connect-NTH17R3] - Calling connect_to_rendezvous_point
2023-03-30 15:31:35,056 - INFO - [Connect-NTH17R3] - Connect to rendezvous point NTH17R3
2023-03-30 15:31:35,132 - INFO - [Connect-NTH17R3] - Loading cached NetworkStatusDocument from TorCacheDirStorage: /root/.local/share/torpy/network_status
2023-03-30 15:31:36,097 - INFO - [Connect-NTH17R3] - Loading cached DirKeyCertificateList from TorCacheDirStorage: /root/.local/share/torpy/dir_key_certificates
2023-03-30 15:31:36,316 - INFO - [Connect-NTH17R3] - Connected to rendezvous point NTH17R3
2023-03-30 15:31:36,322 - INFO - [Connect-NTH17R3] - Waiting for peer to open stream
2023-03-30 15:31:36,392 - INFO - [Connect-NTH17R3] - Received cell: <class 'torpy.cells.CellRelay'>
2023-03-30 15:31:36,393 - INFO - [Connect-NTH17R3] - Stream opened successfully
2023-03-30 15:31:36,393 - INFO - [Connect-NTH17R3] - Sending HELLO to peer: b'\x80\x04\x95&\x00\x00\x00\x00\x00\x00\x00]\x94}\x94(\x8c\x07Message\x94\x8c\x05HELLO\x94\x8c\x04Peer\x94\x8c\x02P2\x94ua.'
2023-03-30 15:31:36,597 - INFO - [RECEIVER] - Added neighbour P1
2023-03-30 15:32:16,705 - INFO - [CONNECTION] - Loading cached NetworkStatusDocument from TorCacheDirStorage: /root/.local/share/torpy/network_status
2023-03-30 15:32:18,471 - INFO - [CONNECTION] - Loading cached DirKeyCertificateList from TorCacheDirStorage: /root/.local/share/torpy/dir_key_certificates
2023-03-30 15:32:18,494 - INFO - [MainThread] - ['ESTABLISH', 'NTH26R3', b'd4d40cdebc36952994ba']
2023-03-30 15:32:18,495 - INFO - [MainThread] - Got ESTABLISH to relay NTH26R3:b'd4d40cdebc36952994ba' via staysafeandgoodluck
2023-03-30 15:32:18,499 - INFO - [Establish-NTH26R3] - Establishing RP to NTH26R3
2023-03-30 15:32:18,590 - INFO - [Establish-NTH26R3] - Loading cached NetworkStatusDocument from TorCacheDirStorage: /root/.local/share/torpy/network_status
2023-03-30 15:32:19,617 - INFO - [Establish-NTH26R3] - Loading cached DirKeyCertificateList from TorCacheDirStorage: /root/.local/share/torpy/dir_key_certificates
2023-03-30 15:32:19,621 - INFO - [Establish-NTH26R3] - Connecting to guard node 185.181.61.23:443 (staysafeandgoodluck; Tor 0.4.7.10)... (None)
2023-03-30 15:32:19,723 - INFO - [Establish-NTH26R3] - Creating new circuit #80000006 with 185.181.61.23:443 (staysafeandgoodluck; Tor 0.4.7.10) router...
2023-03-30 15:32:19,749 - INFO - [Establish-NTH26R3] - Building -1 hops circuit...
2023-03-30 15:32:19,751 - INFO - [Establish-NTH26R3] - Extending the circuit #80000006 with 192.42.116.178:9002 (NTH26R3; Tor 0.4.7.13)...
2023-03-30 15:32:19,752 - INFO - [Establish-NTH26R3] - Getting descriptor for 192.42.116.178:9002 (NTH26R3; Tor 0.4.7.13)...
2023-03-30 15:32:19,771 - INFO - [Establish-NTH26R3] - Connecting to guard node 45.61.186.203:9001 (6517357147976530; Tor 0.4.7.10)... (Internal dir client)
2023-03-30 15:32:20,503 - INFO - [Establish-NTH26R3] - Creating new circuit #80000007 with 45.61.186.203:9001 (6517357147976530; Tor 0.4.7.10) router...
2023-03-30 15:32:20,835 - INFO - [Establish-NTH26R3] - Building 0 hops circuit...
2023-03-30 15:32:20,835 - INFO - [Establish-NTH26R3] - Stream #7: creating attached to #80000007 circuit...
2023-03-30 15:32:20,835 - INFO - [Establish-NTH26R3] - Stream #7: connecting to hsdir
2023-03-30 15:32:20,991 - INFO - [RecvLoop_45.61.1] - Stream #7: connected (remote ip '')
2023-03-30 15:32:21,344 - INFO - [RecvLoop_45.61.1] - Stream #7: remote disconnected (reason = DONE)
2023-03-30 15:32:21,344 - INFO - [Establish-NTH26R3] - Stream #7: closing (state = Disconnected)...
2023-03-30 15:32:21,344 - INFO - [Establish-NTH26R3] - Got descriptor
2023-03-30 15:32:21,495 - INFO - [Establish-NTH26R3] - <class 'torpy.circuit.TorCircuit'>
2023-03-30 15:32:21,542 - INFO - [Establish-NTH26R3] - Rendezvous established (CellRelayRendezvousEstablished())
2023-03-30 15:32:21,542 - INFO - [Establish-NTH26R3] - Waiting for connections at relay NTH26R3 for cookie b'd4d40cdebc36952994ba' ...
2023-03-30 15:32:28,504 - INFO - [CONNECTION] - Sending JOIN to P4 for relay NTH26R3 with cookie b'd4d40cdebc36952994ba'
2023-03-30 15:32:30,200 - INFO - [Establish-NTH26R3] - Got REND2 message from NTH26R3
2023-03-30 15:32:30,201 - INFO - [Establish-NTH26R3] - Entered circuit context with id 0x80000006
2023-03-30 15:32:30,201 - INFO - [Establish-NTH26R3] - Stream #8: creating attached to #80000006 circuit...
2023-03-30 15:32:30,201 - INFO - [Establish-NTH26R3] - Stream #8: connecting to ('127.0.0.1', 5000)
2023-03-30 15:32:30,275 - INFO - [RecvLoop_185.181] - Stream #8: connected (remote ip '127.0.0.1')
2023-03-30 15:32:30,404 - INFO - [Establish-NTH26R3] - Sending HELLO to peer
2023-03-30 15:32:30,404 - INFO - [Establish-NTH26R3] - Sending HELLO to peer: b'\x80\x04\x95&\x00\x00\x00\x00\x00\x00\x00]\x94}\x94(\x8c\x07Message\x94\x8c\x05HELLO\x94\x8c\x04Peer\x94\x8c\x02P2\x94ua.'
2023-03-30 15:32:30,406 - INFO - [RECEIVER] - Added neighbour P4
2023-03-30 15:32:36,312 - WARNING - [Connect-NTH17R3] - Got SENDME
2023-03-30 15:32:36,321 - ERROR - [Connect-WinstonSmith] - Got RELAY_END
2023-03-30 15:32:51,067 - WARNING - [RECEIVER] - Routing loop from P2 to P3: P2-P4-P2
2023-03-30 15:33:29,332 - ERROR - [Establish-IdeaNexus] - recv timeout
2023-03-30 15:33:29,334 - INFO - [Establish-IdeaNexus] - Closing connection Establish-IdeaNexus
2023-03-30 15:33:29,334 - INFO - [Establish-IdeaNexus] - Stream #6: closing (state = Connected)...
2023-03-30 15:33:29,335 - INFO - [Establish-IdeaNexus] - Destroy circuit #80000004
2023-03-30 15:33:29,337 - INFO - [Establish-IdeaNexus] - Closing guard connections (None)...
2023-03-30 15:33:31,957 - INFO - [RecvLoop_45.61.1] - On destroy: circuit #80000007
2023-03-30 15:33:31,958 - INFO - [RecvLoop_45.61.1] - Destroy circuit #80000007
2023-03-30 15:33:44,761 - WARNING - [Connect-NTH17R3] - Got SENDME
2023-03-30 15:33:44,824 - WARNING - [Connect-NTH17R3] - Got SENDME
2023-03-30 15:34:46,546 - WARNING - [Connect-NTH17R3] - Got SENDME
2023-03-30 15:35:51,692 - WARNING - [Connect-NTH17R3] - Got SENDME
2023-03-30 15:35:51,744 - WARNING - [Connect-NTH17R3] - Got SENDME
2023-03-30 15:37:01,822 - WARNING - [Connect-NTH17R3] - Got SENDME
2023-03-30 15:37:46,781 - INFO - [RECEIVER] - Added relay Chimera from P1 to JOIN queue
2023-03-30 15:37:46,782 - INFO - [MainThread] - ['JOIN', 'Chimera', b'76e4cafe2015a0af45ee']
2023-03-30 15:37:46,782 - INFO - [MainThread] - Got JOIN to relay Chimera, waiting 10 sec to connect
2023-03-30 15:37:46,798 - INFO - [Connect-Chimera] - Calling connect_to_rendezvous_point
2023-03-30 15:37:46,798 - INFO - [Connect-Chimera] - Connect to rendezvous point Chimera
2023-03-30 15:37:46,844 - INFO - [Connect-Chimera] - Loading cached NetworkStatusDocument from TorCacheDirStorage: /root/.local/share/torpy/network_status
2023-03-30 15:37:48,395 - INFO - [Connect-Chimera] - Loading cached DirKeyCertificateList from TorCacheDirStorage: /root/.local/share/torpy/dir_key_certificates
2023-03-30 15:37:49,058 - INFO - [Connect-Chimera] - Connected to rendezvous point Chimera
2023-03-30 15:37:49,062 - INFO - [Connect-Chimera] - Waiting for peer to open stream
2023-03-30 15:37:49,319 - INFO - [Connect-Chimera] - Received cell: <class 'torpy.cells.CellRelay'>
2023-03-30 15:37:49,323 - INFO - [Connect-Chimera] - Stream opened successfully
2023-03-30 15:37:49,323 - INFO - [Connect-Chimera] - Sending HELLO to peer: b'\x80\x04\x95&\x00\x00\x00\x00\x00\x00\x00]\x94}\x94(\x8c\x07Message\x94\x8c\x05HELLO\x94\x8c\x04Peer\x94\x8c\x02P2\x94ua.'
2023-03-30 15:37:49,738 - INFO - [RECEIVER] - Added neighbour P1
2023-03-30 15:38:28,884 - INFO - [CONNECTION] - Loading cached NetworkStatusDocument from TorCacheDirStorage: /root/.local/share/torpy/network_status
2023-03-30 15:38:30,810 - INFO - [CONNECTION] - Loading cached DirKeyCertificateList from TorCacheDirStorage: /root/.local/share/torpy/dir_key_certificates
2023-03-30 15:38:30,858 - INFO - [MainThread] - ['ESTABLISH', '6329cogwheel', b'1a8a92609aedd0b14ab9']
2023-03-30 15:38:30,858 - INFO - [MainThread] - Got ESTABLISH to relay 6329cogwheel:b'1a8a92609aedd0b14ab9' via staysafeandgoodluck
2023-03-30 15:38:30,858 - INFO - [Establish-6329cogwheel] - Establishing RP to 6329cogwheel
2023-03-30 15:38:30,968 - INFO - [Establish-6329cogwheel] - Loading cached NetworkStatusDocument from TorCacheDirStorage: /root/.local/share/torpy/network_status
2023-03-30 15:38:33,202 - INFO - [Establish-6329cogwheel] - Loading cached DirKeyCertificateList from TorCacheDirStorage: /root/.local/share/torpy/dir_key_certificates
2023-03-30 15:38:33,216 - INFO - [Establish-6329cogwheel] - Connecting to guard node 185.181.61.23:443 (staysafeandgoodluck; Tor 0.4.7.10)... (None)
2023-03-30 15:38:33,293 - INFO - [Establish-6329cogwheel] - Creating new circuit #80000008 with 185.181.61.23:443 (staysafeandgoodluck; Tor 0.4.7.10) router...
2023-03-30 15:38:33,319 - INFO - [Establish-6329cogwheel] - Building -1 hops circuit...
2023-03-30 15:38:33,326 - INFO - [Establish-6329cogwheel] - Extending the circuit #80000008 with 114.34.165.141:443 (6329cogwheel; Tor 0.4.7.13)...
2023-03-30 15:38:33,329 - INFO - [Establish-6329cogwheel] - Getting descriptor for 114.34.165.141:443 (6329cogwheel; Tor 0.4.7.13)...
2023-03-30 15:38:33,340 - INFO - [Establish-6329cogwheel] - Connecting to guard node 194.147.140.104:8443 (tor104IPConnectINFO; Tor 0.4.7.11)... (Internal dir client)
2023-03-30 15:38:33,587 - INFO - [Establish-6329cogwheel] - Creating new circuit #80000009 with 194.147.140.104:8443 (tor104IPConnectINFO; Tor 0.4.7.11) router...
2023-03-30 15:38:33,668 - INFO - [Establish-6329cogwheel] - Building 0 hops circuit...
2023-03-30 15:38:33,668 - INFO - [Establish-6329cogwheel] - Stream #9: creating attached to #80000009 circuit...
2023-03-30 15:38:33,668 - INFO - [Establish-6329cogwheel] - Stream #9: connecting to hsdir
2023-03-30 15:38:33,729 - INFO - [RecvLoop_194.147] - Stream #9: connected (remote ip '')
2023-03-30 15:38:33,956 - INFO - [RecvLoop_194.147] - Stream #9: remote disconnected (reason = DONE)
2023-03-30 15:38:33,960 - INFO - [Establish-6329cogwheel] - Stream #9: closing (state = Disconnected)...
2023-03-30 15:38:33,960 - INFO - [Establish-6329cogwheel] - Got descriptor
2023-03-30 15:38:34,362 - INFO - [Establish-6329cogwheel] - <class 'torpy.circuit.TorCircuit'>
2023-03-30 15:38:34,679 - INFO - [Establish-6329cogwheel] - Rendezvous established (CellRelayRendezvousEstablished())
2023-03-30 15:38:34,679 - INFO - [Establish-6329cogwheel] - Waiting for connections at relay 6329cogwheel for cookie b'1a8a92609aedd0b14ab9' ...
2023-03-30 15:38:40,866 - INFO - [CONNECTION] - Sending JOIN to P4 for relay 6329cogwheel with cookie b'1a8a92609aedd0b14ab9'
2023-03-30 15:38:44,336 - INFO - [Establish-6329cogwheel] - Got REND2 message from 6329cogwheel
2023-03-30 15:38:44,341 - INFO - [Establish-6329cogwheel] - Entered circuit context with id 0x80000008
2023-03-30 15:38:44,341 - INFO - [Establish-6329cogwheel] - Stream #10: creating attached to #80000008 circuit...
2023-03-30 15:38:44,341 - INFO - [Establish-6329cogwheel] - Stream #10: connecting to ('127.0.0.1', 5000)
2023-03-30 15:38:44,939 - INFO - [RecvLoop_185.181] - Stream #10: connected (remote ip '127.0.0.1')
2023-03-30 15:38:44,955 - INFO - [Establish-6329cogwheel] - Sending HELLO to peer
2023-03-30 15:38:44,955 - INFO - [Establish-6329cogwheel] - Sending HELLO to peer: b'\x80\x04\x95&\x00\x00\x00\x00\x00\x00\x00]\x94}\x94(\x8c\x07Message\x94\x8c\x05HELLO\x94\x8c\x04Peer\x94\x8c\x02P2\x94ua.'
2023-03-30 15:38:45,243 - INFO - [RECEIVER] - Added neighbour P4
2023-03-30 15:38:49,093 - ERROR - [Connect-NTH17R3] - Got RELAY_END
2023-03-30 15:38:57,361 - WARNING - [Connect-Chimera] - Got SENDME
2023-03-30 15:39:43,727 - INFO - [RecvLoop_194.147] - On destroy: circuit #80000009
2023-03-30 15:39:43,732 - INFO - [RecvLoop_194.147] - Destroy circuit #80000009
2023-03-30 15:39:45,053 - ERROR - [Establish-NTH26R3] - recv timeout
2023-03-30 15:39:45,054 - INFO - [Establish-NTH26R3] - Closing connection Establish-NTH26R3
2023-03-30 15:39:45,055 - INFO - [Establish-NTH26R3] - Stream #8: closing (state = Connected)...
2023-03-30 15:39:45,064 - INFO - [Establish-NTH26R3] - Destroy circuit #80000006
2023-03-30 15:39:45,066 - INFO - [Establish-NTH26R3] - Closing guard connections (None)...
2023-03-30 15:39:57,513 - WARNING - [Connect-Chimera] - Got SENDME
2023-03-30 15:39:57,634 - WARNING - [Connect-Chimera] - Got SENDME
2023-03-30 15:41:13,946 - WARNING - [Connect-Chimera] - Got SENDME
2023-03-30 15:41:52,280 - INFO - [RECEIVER] - updateGraphOnly: [['P2', 'P4', 931, '6329cogwheel'], ['P3', 'P4', 159, 'onionDAOrel0aded1'], ['P1', 'P2', 309, 'Chimera'], ['P1', 'P3', 100, 'definitelyNotLE']] , {'P1'}
2023-03-30 15:42:13,454 - INFO - [RECEIVER] - updateGraphOnly: [['P2', 'P4', 931, '6329cogwheel'], ['P3', 'P4', 159, 'onionDAOrel0aded1']] , {'P3'}
2023-03-30 15:42:27,967 - WARNING - [Connect-Chimera] - Got SENDME
2023-03-30 15:42:28,086 - WARNING - [Connect-Chimera] - Got SENDME
2023-03-30 15:43:08,653 - ERROR - [RecvLoop_185.181] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-30 15:43:10,383 - ERROR - [RecvLoop_185.181] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-30 15:43:10,721 - ERROR - [RecvLoop_185.181] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-30 15:43:11,043 - ERROR - [RecvLoop_185.181] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-30 15:43:13,347 - ERROR - [RecvLoop_185.181] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-30 15:43:15,212 - ERROR - [RecvLoop_185.181] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-30 15:43:15,936 - ERROR - [RecvLoop_185.181] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-30 15:43:17,348 - ERROR - [RecvLoop_185.181] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-30 15:43:17,760 - ERROR - [RecvLoop_185.181] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-30 15:43:18,353 - ERROR - [RecvLoop_185.181] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-30 15:43:18,645 - ERROR - [RecvLoop_185.181] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-30 15:43:18,649 - ERROR - [RecvLoop_185.181] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-30 15:43:18,682 - ERROR - [RecvLoop_185.181] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-30 15:43:20,945 - ERROR - [RecvLoop_185.181] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-30 15:43:23,362 - ERROR - [RecvLoop_185.181] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-30 15:43:25,549 - ERROR - [RecvLoop_185.181] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-30 15:43:25,958 - ERROR - [RecvLoop_185.181] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-30 15:43:26,376 - ERROR - [RecvLoop_185.181] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-30 15:43:26,378 - ERROR - [RecvLoop_185.181] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-30 15:43:26,485 - INFO - [Thread-4] - Adding P4 to list of inactive
2023-03-30 15:43:26,485 - INFO - [Thread-4] - DEAD ROUTES DETECTED: P4
2023-03-30 15:43:26,486 - INFO - [Thread-4] - Removing P4 from neighbours
2023-03-30 15:43:26,486 - INFO - [Thread-4] - Removed circuit for P4
2023-03-30 15:43:26,486 - INFO - [Thread-4] - updateGraphOnly: [['P2', 'P4', 931, '6329cogwheel'], ['P3', 'P4', 159, 'onionDAOrel0aded1'], ['P1', 'P2', 534, 'Chimera']] , {'P4'}
2023-03-30 15:43:27,494 - INFO - [Thread-4] - GRAPH: [['P1', 'P2', 534, 'Chimera']]
2023-03-30 15:43:27,495 - INFO - [Thread-4] - ROUTER: {'RID': 'P2', 'Port': 5002, 'Neighbours': 1, 'Neighbours Data': [{'NID': 'P1', 'Cost': 514, 'Hostname': '127.0.0.1', 'RP': 'Chimera', 'FLAG': 0}], 'SN': 24, 'FLAG': 0, 'RP': set()}
2023-03-30 15:43:27,495 - INFO - [Thread-4] - LSA_DB: {'P1': {'RID': 'P1', 'Port': 5001, 'Neighbours': 1, 'Neighbours Data': [{'NID': 'P2', 'Cost': 534, 'Hostname': '127.0.0.1', 'RP': 'Chimera', 'FLAG': 0}], 'SN': 25, 'FLAG': 1, 'RP': set()}}
2023-03-30 15:43:29,311 - WARNING - [Connect-Chimera] - Got SENDME
2023-03-30 15:43:30,261 - ERROR - [RecvLoop_185.181] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-30 15:43:30,973 - ERROR - [RecvLoop_185.181] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-30 15:43:36,412 - ERROR - [RecvLoop_185.181] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-30 15:43:36,484 - ERROR - [RecvLoop_185.181] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-30 15:43:36,486 - ERROR - [RecvLoop_185.181] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-30 15:43:36,748 - ERROR - [RecvLoop_185.181] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-30 15:43:36,751 - ERROR - [RecvLoop_185.181] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-30 15:43:40,451 - ERROR - [RecvLoop_185.181] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-30 15:43:40,791 - ERROR - [RecvLoop_185.181] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-30 15:43:41,112 - ERROR - [RecvLoop_185.181] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-30 15:43:41,122 - INFO - [CONNECTION] - Neighbour connections (1) less than MIN_NEIGHBOUR_CONNECTIONS (2)
2023-03-30 15:43:41,122 - INFO - [CONNECTION] - Candidate peers: []
2023-03-30 15:43:41,122 - INFO - [CONNECTION] - Highest cost peers: [('P3', 1090), ('P4', 931), ('P1', 534), ('P2', 0)]
2023-03-30 15:43:41,122 - INFO - [CONNECTION] - No join peers found
2023-03-30 15:43:41,443 - ERROR - [RecvLoop_185.181] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-30 15:43:41,456 - ERROR - [RecvLoop_185.181] - Some handle errors
Traceback (most recent call last):
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 224, in _do_recv
    self._handler_mgr.handle(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 280, in handle
    handler(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 43, in wrapped
    return func(*args_new, **kwargs)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/guard.py", line 127, in _on_relay
    circuit.handle_relay(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 456, in handle_relay
    circuit_node, inner_cell = self._decrypt(cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/circuit.py", line 522, in _decrypt
    return from_node, relay_cell.get_decrypted()
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cells.py", line 234, in get_decrypted
    assert self._checked
AssertionError
2023-03-30 15:43:52,253 - ERROR - [Connect-Chimera] - Got DESTROY
